packages/x-date-pickers/src/DateCalendar/DayCalendar.tsx

```diff
@@ -75,6 +76,11 @@ export interface ExportedDayCalendarProps extends ExportedPickersDayProps {
    * Put it to 6 to have a fixed number of weeks in Gregorian calendars
    */
   fixedWeekNumber?: number;
+  /**
+   * If `true`, day slot will use the v8 version.
```

Comment:
Not a big fan of "v8 version" here.
But we have a broader question of how we brand this new feature (the name of the `enableEnhancedDaySlot` prop)
For the field, we went for "accessible" instead of "v7" or "new" or "enhanced".
For this day component, we could go for "customizable" since it's the main benefit. Of "withSimpleDOMStructure" but it's weird :laughing: 

cc @LukasTy @joserodolfofreitas